db/db_test.go

```diff
@@ -76,24 +93,37 @@ func TestDel(t *testing.T) {
 	db, err := Load(sp)
 	assert.NoError(err)
 
-	err = db.Put("foo", []byte(`"bar"`))
+	wtx := db.NewTransaction()
```

Comment:
I'll leave it to you, but I've really come to appreciate the go-style table driven tests. For things like this that we really care about every branch being tested and working per-design, it's easy to prove to yourself that you've covered them all.

You'd take each individual method of `Transaction` and create a test struct that has a field for each case where the code branches. So for like `Transaction::Put`:

```
func TestTransactionPut(t *testing.T) {
...
tc := []struct {
    txClosed bool
    path string
    json string
    expectedError string
}{
    {true, "foo", "42", ErrorClosed.Error()},
    {false, "", "42", "Empty ID not supported"},
    {false, "foo", "", "Invalid JSON"},
    ...
}
```

Then just create an entry that exercises each branch.

Reply:
This one used to be a lot more subtle when there we read and write transactions.

Reply:
One thing I really do not like about table driven tests (the way we write them) is that it is hard to see which test failed. I thought that `t.Run` was supposed to help but it doesn't (it does allow running subtests which is nice)